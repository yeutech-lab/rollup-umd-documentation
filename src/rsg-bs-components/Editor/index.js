export { default } from 'rsg-components/Editor/EditorLoader';
