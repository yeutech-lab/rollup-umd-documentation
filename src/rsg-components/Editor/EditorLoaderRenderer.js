import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';

const styles = ({ fontFamily, color, space, fontSize }) => ({
	root: {
		padding: [[space[1], space[2], space[1], space[1]]],
		fontFamily: fontFamily.base,
		fontSize: fontSize.small,
		color: color.light,
		backgroundColor: color.codeBackground,
	},
	// Tweak CodeMirror styles. Duplicate selectors are for increased specificity
	'@global': {
		'.CodeMirror.CodeMirror': {
			isolate: false,
			fontFamily: fontFamily.monospace,
			height: 'auto',
			padding: [[space[1], space[2]]],
			fontSize: fontSize.small,
		},
		'.CodeMirror.CodeMirror pre': {
			isolate: false,
			padding: 0,
		},
		'.CodeMirror-scroll.CodeMirror-scroll': {
			isolate: false,
			height: 'auto',
			overflowY: 'hidden',
			overflowX: 'auto',
		},
		'.cm-error.cm-error': {
			isolate: false,
			background: 'none',
		},
	},
});

export function EditorLoaderRenderer({ classes }) {
	return <div className={classes.root}>Loadingâ€¦</div>;
}

EditorLoaderRenderer.propTypes = {
	classes: PropTypes.object.isRequired,
};

export default Styled(styles)(EditorLoaderRenderer);
