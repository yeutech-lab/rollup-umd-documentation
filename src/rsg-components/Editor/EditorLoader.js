import React, { Component } from 'react';
import EditorLoaderRenderer from 'rsg-components/Editor/EditorLoaderRenderer';

export default class EditorLoader extends Component {
	state = {
		editor: null,
	};

	componentDidMount() {
		import('rsg-components/Editor/Editor').then(module => {
			this.setState({ editor: module.default });
		});
	}

	render() {
		const Editor = this.state.editor;
		if (Editor) {
			return <Editor {...this.props} />;
		}

		return <EditorLoaderRenderer />;
	}
}
