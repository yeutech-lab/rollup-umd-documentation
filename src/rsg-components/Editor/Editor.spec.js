import React from 'react';
import PropTypes from 'prop-types';
import { EditorLoaderRenderer } from './EditorLoaderRenderer';
import Editor from './Editor';

const code = '<button>MyAwesomeCode</button>';
const newCode = '<button>MyNewAwesomeCode</button>';
const options = {
	context: {
		config: {
			showCode: false,
			highlightTheme: 'base16-light',
			editorConfig: {
				mode: 'js',
			},
		},
	},
	childContextTypes: {
		config: PropTypes.object.isRequired,
	},
};

describe('EditorLoaderRenderer', () => {
	it('should renderer should render loader', () => {
		const actual = shallow(<EditorLoaderRenderer classes={{}} />);

		expect(actual).toMatchSnapshot();
	});
});

describe('Editor', () => {
	it('should renderer and editor', () => {
		const actual = shallow(<Editor code={code} onChange={() => {}} />, options);

		expect(actual).toMatchSnapshot();
	});

	it('should update code with debounce', done => {
		const onChange = jest.fn();
		const actual = mount(<Editor code={code} onChange={onChange} />, options);

		expect(actual.text()).toMatch(code);

		// Set new value by calling a method on the CodeMirror instance
		actual
			.find('div')
			.getDOMNode()
			.querySelector('.CodeMirror')
			.CodeMirror.setValue(newCode);

		setTimeout(() => {
			expect(onChange).toBeCalledWith(newCode);
			done();
		}, 13);
	});
});
