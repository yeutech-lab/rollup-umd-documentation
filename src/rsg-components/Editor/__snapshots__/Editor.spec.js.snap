// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Editor should renderer and editor 1`] = `
<UnControlled
  onChange={[Function]}
  options={
    Object {
      "mode": "js",
    }
  }
  value=
<button>
  MyAwesomeCode
</button>

/>
`;

exports[`EditorLoaderRenderer should renderer should render loader 1`] = `
<div>
  Loadingâ€¦
</div>
`;
